### PR TITLE
2022 Febuary Ports - Part 2

### PR DESCRIPTION
**Pull request by Jamini**
## About The Pull Request

This ports the ghost role system from eris to our repo.
No other major changes have been made in it thus far, but the framework is there

## Why It's Good For The Game

Upstream ports

## Changelog
```changelog
add: Ghostrole system
```

### CODE REVIEW

File: code/game/objects/effects/spawners/ghostroles.dm [hunk 1]
```diff
@@ -0,0 +1,70 @@
+//Prisoner containment sleeper: Ghosts become escaped prisoners and are advised to find a way out of the mess they've gotten themselves into.
+/obj/effect/mob_spawn/human/prisoner_transport
+	name = "prisoner containment sleeper"
+	desc = "A sleeper designed to put its occupant into a deep coma, unbreakable until the sleeper turns off. This one's glass is cracked and you can see a pale, sleeping face staring out."
+	mob_name = "an escaped prisoner"
+	icon = 'icons/obj/Cryogenic2.dmi'
+	icon_state = "sleeper_1"
+	outfit = /decl/hierarchy/outfit/escapedprisoner
+	short_desc = "You're a prisoner, sentenced to hard work in one of Ironhammer's labor camps, but it seems as \
```
Comment (Shadow-Quill) [inline code/game/objects/effects/spawners/ghostroles.dm:9]: `Ironhammer`

File: maps/encounters/prisonhulk/prisonhulk.dmm [hunk 2]
```diff
@@ -0,0 +1,5 @@
+"a" = ()
```
Comment (Shadow-Quill) [inline maps/encounters/prisonhulk/prisonhulk.dmm:1]: This doesn't seem quite right.

**Review by Shadow-Quill — CHANGES_REQUESTED** [review]

File: code/game/objects/effects/spawners/ghostroles.dm (hunk 1 above)
Comment (Jamini) [reply code/game/objects/effects/spawners/ghostroles.dm:9]: We'll do a lore pass later. I'd rather not change it in this file

File: maps/encounters/prisonhulk/prisonhulk.dmm (hunk 2 above)
Comment (Jamini) [reply maps/encounters/prisonhulk/prisonhulk.dmm:1]: Can you clarify?

Comment (Shadow-Quill) [reply maps/encounters/prisonhulk/prisonhulk.dmm:1]: This is an empty map -- though I'm not sure if it was intentional.

Comment (Jamini) [reply maps/encounters/prisonhulk/prisonhulk.dmm:1]: Yeah. We're going to need to make our own. Later.

**Review by Shadow-Quill — APPROVED** [review]

**Pull request merged**